src/lib/Syntax.hs

```diff
@@ -2396,6 +2418,32 @@ instance Store (Effect n)
 instance Store (DataConRefBinding n l)
 instance NameColor c => Store (Binding c n)
 
+instance Hashable ForAnn
+instance Hashable AddressSpace
+instance Hashable LetAnn
+instance Hashable RWS
+instance Hashable Direction
+instance Hashable UnOp
+instance Hashable BinOp
+instance Hashable CmpOp
+instance Hashable BaseType
+instance Hashable LitVal
+instance Hashable ScalarBaseType
+instance Hashable Device
+instance Hashable Arrow
+
+instance Hashable IsCUDARequired
```

Comment:
Why do we need to hash those? 🤔 Is that for future changes?

Reply:
I just made `AlphaHashableE` instances for anything that previously had `AlphaEqE`. These instances were required for the Imp data types. How knows, maybe we'll eventually want to CSE at the Imp level too.